blog/content/post/2020-06-30-trenchboot-grub-cbfs.md

```diff
@@ -0,0 +1,575 @@
+    ipit a, scelerisque sed, lacinia in, mi. Cras vel lorem. Etiam pellentesque ali
+    quet tellus. Phasellus pharetra nullla ac diam. Quisque semper justo at risus. D
+    onec venenatis, turpis vel hendrerit interdum, dui ligula ultricies purus, sed
+    posuere libero dui id orci. Nam congue, pede vitae dapibus aliquet, elit magna
+    vulputate arcu, vel tempus metus leo non est. Etiam sit amet lectus quis est co
+    ngue mollis. Phasellus congue lacus eget neque. Phasellus ornare, ante vitae co
+    nsectetuer consequat, purus sapien ultricies dolor, et mollis pede metus eget n
+    isi. Praesent sodales velit quis augue. Cras suscipit, urna at aliquuam rhoncus,
+    urna quam viverra nisi, in interdum massa nibh nec erat.
+    ```
+
+    If content of above command is exactly the same as your `test-file.txt`,
+    than it means that **GRUB supports LZMA**
+
+## TPM event log
+
+Recently, we have possibility to verify firmware by analyzing PCRs values.
+However, there is still lack of information about TPM itself. User doesn't have
+any details about operations which are done, especially hashes of every measured
+component. Without them, it is hard to make an attestation and fully trust that
+firmware is not malicious. As we see this deficiency, we introduced **TPM event
+log**. Idea is very simple. Every event related to TPM is registered. It means
+that every operation done by TPM is memorized and can be accessed by user at any
+time. Most interesting parts are of course **hashes of every firmware
+component**. With this knowledge and details about PCR extension process
+(extension order mostly) user have tool to analyze them, if there is a need,
+verify against corruption. All you have to do is to make your own calculations
+and see if every step is exactly as TPM event log reports.
+
+As you can see, it is a great security feature because now user has possibility
+to attest TPM. In case of calculations incompatibility, it also allows to easily
+track corrupted part of firmware. Now, when you know what is TPM event log and
+we believe you see its great benefit, let's see how it works and how to use it.
+
+### TPM event log verification
+
+Hardware and firmware specification which we are using for test:
+
+- PC Engines apu2 platform with TPM2.0 or TPM1.2
+- coreboot v4.12.0.2
+- NixOS operating system
+
+>We don't ensure valid test results if you don't have exactly the same
+configuration and environment as above.
+
+Following procedure will guide you step-by-step how to enable and read TPM event
+log. Before giving final example, you need to prepare all necessary components.
+
+1. Install `xxd` tool.
+
+    > This tool will be used during `landing-zone` build, so it must be done
+    earlier.
+
+    ```
+    $ cd ~/nixpkgs
+    $ nix-build -A unixtools.xxd
+    (...)
+    /nix/store/2q94zc1agpkvchxxnx6pwy1v6rpdqzdx-xxd-1003.1-2008
+    ```
+
+    Last line points to directory where package is installed. Copy `/bin/xxd`
```

Comment:
Is this really necessary? Every sane build system would include some kind of 'install' operation.

Reply:
It is necessary if we want to have xxd. I didn't investigate why it is missed in base install and don't have really time for that